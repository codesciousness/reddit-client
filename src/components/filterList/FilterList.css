.FilterList {
    width: 100%;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    margin-top: 2rem;
}

.FilterList__prompt {
    font-size: 1.25rem;
    margin: 2rem;
}

.FilterList__container {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    width: 75%;
    margin: auto;
}