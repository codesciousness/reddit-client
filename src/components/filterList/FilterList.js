import React from 'react';
import './FilterList.css';
import Filter from '../../features/filter/Filter';

const flairs = ['Animal Science', 'Anthropology', 'Astronomy', 'Biology', 'Cancer', 'Chemistry', 'Computer Science', 'Earth Science', 'Economics', 'Engineering', 'Environment', 'Epidemiology', 'Geology', 'Health', 'Mathematics', 'Medicine', 'Nanoscience', 'Neuroscience', 'Paleontology', 'Physics', 'Psychology', 'Social Science'];

const FilterList = () => {
    return (
        <section className="FilterList">
            <p className="FilterList__prompt">What interests you?</p>
            <div className="FilterList__container">
                {flairs.map(flair => {
                    return <Filter flair={flair} />
                })}
            </div>
        </section>
    );
}

export default FilterList;