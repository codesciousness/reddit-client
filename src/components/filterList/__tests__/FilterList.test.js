import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import FilterList from '../FilterList';

jest.mock('../../../features/filter/Filter', () => () => (<button>Test Filter</button>));

describe('<FilterList />', () => {
    test('FilterList component is defined', () => {
        expect(<FilterList />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(<FilterList />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly', () => {
        render(<FilterList />);
        expect(screen.getByText('Your Daily Dose of Science')).toBeInTheDocument();
        expect(screen.getByText('What interests you?')).toBeInTheDocument();
    });
    test('renders 22 Filter components', () => {
        render(<FilterList />);
        expect(screen.getAllByText('Test Filter')).toHaveLength(22);
    });
});