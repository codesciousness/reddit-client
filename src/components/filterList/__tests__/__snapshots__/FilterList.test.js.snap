// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterList /> matches the snapshot 1`] = `
<section
  className="FilterList"
>
  <h2
    className="FilterList__title"
  >
    Your Daily Dose of Science
  </h2>
  <p
    className="FilterList__prompt"
  >
    What interests you?
  </p>
  <ul
    className="FilterList__container"
  >
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
    <li>
      <button>
        Test Filter
      </button>
    </li>
  </ul>
</section>
`;
